lightgbm/src/main/scala/com/microsoft/ml/spark/lightgbm/SharedState.scala

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark.lightgbm
+
+import java.util.concurrent.CountDownLatch
+
+import com.microsoft.ml.spark.lightgbm.dataset.DatasetAggregator
+import com.microsoft.ml.spark.lightgbm.params.TrainParams
+import org.apache.spark.sql.types.StructType
+import org.slf4j.Logger
+
+class SharedState {
+  var mainExecutorWorker: Long = LightGBMUtils.getTaskId()
+
+  @volatile var datasetAggregator: Option[DatasetAggregator] = None
+  def linkDatasetAggregator(columnParams: ColumnParams,
```

Comment:
For linkDatasetAggregator and linkIsSparse is there any way we can make this stuff in the constructor of the shared state? for linkDatasetAggregator it seems it only takes parameters from the head node anyway so that seems possible

Reply:
tried this for datasetAggregator, will see if it works

Reply:
Did you push these changes i dont see anything in the constructor of SharedState

Reply:
yes I did the constructor looks like this now:
```
class SharedState(columnParams: ColumnParams,
                  schema: StructType,
                  trainParams: TrainParams) {
```